samples/performance/README.md

```diff
@@ -0,0 +1,78 @@
+# Gateway Performance Measurement
+
+Measure performance of the Gateway SDK.
+
+## Overview
+
+Sample goals:
+- Ensure correctness under stress.
+- Measure latency through gateway.
+- Measure throughput over time.
+
+There are two modules as part of this sample.  There is a simulator module 
+producing messages and a metrics module consuming messages. The modules are 
+coupled by the message contents and the gateway links. The following is a 
+generic overview of the sample gateway:
+
+![Overview](Overview.png) 
+
+## Metrics gathering
+
+The metrics module assumes some information is present in the message properties 
+in order to generate the desired metrics. The following table contains the 
+expected message properties:
+
+| Property           | Description    |
+| ------------------ | -------------- |
+| "deviceId"         | A device ID, assumed to be unique for each simulator module |
+| "sequence number"  | The number of the message produced. |
+| "timestamp"        | Time since clock epoch, in microseconds |
+
+When the gateway shuts down, all metrics modules will log statistics for 
+messages received.
+
+The information the metrics module produces is:
+- Test Duration in seconds. (Time from module start to module destroy.)
+- Messages received.
+- Messages per second.
+- Average latency (in microseconds).
+- Maximum latency (in mocroseconds).
+- Number of non-conforming messages.
+- Number of devices discovered.
+- Per device statistics:
+    - Messages received.
+    - Number of out-of-sequence messages. 
```

Comment:
I should. I was worried how accurate I can make it if we receive message in the incorrect order, but I can add it.